### PR TITLE
Setting up Containerization with Docker

### LINKED ISSUE #5: Containerization with Docker

### PR DESCRIPTION
**Pull request by Milosz-cat**
Based on official website: [/building-multi-container-net-app-using-docker-desktop](https://www.docker.com/blog/building-multi-container-net-app-using-docker-desktop/)

`dataprotectionkeys` file is generated in dotnet project automatically and we need to add volumen to handle it ([how-to-store-data-protection-keys](https://stackoverflow.com/questions/61452280/how-to-store-data-protection-keys-outside-of-the-docker-container))

Run `docker compose up --build`
Then Paste `http://localhost:5057/` in your browser

closes #5

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM mcr.microsoft.com/dotnet/sdk as build
+COPY . ./src
+WORKDIR /src
+RUN dotnet build "SzczurTv/SzczurTv.sln" -o /app
+RUN dotnet publish "SzczurTv/SzczurTv.sln" -o /publish
```
Comment (pietrykovsky) [inline Dockerfile:5]: we should dockerize the development version first, we don't need the production right now, we can add it later. this way we won't be blocked with development

File: dataprotectionkeys/key-f2775ffb-7be2-4281-868c-e08023256253.xml
Comment (pietrykovsky) [inline dataprotectionkeys/key-f2775ffb-7be2-4281-868c-e08023256253.xml:1]: I'm not sure if it is safe to have this file in the repository, isn't it some kind of secret key that should be protected?

Comment (Milosz-cat) [reply dataprotectionkeys/key-f2775ffb-7be2-4281-868c-e08023256253.xml:1]: i will take care of it in another issue

File: .vscode/settings.json
Comment (pietrykovsky) [inline .vscode/settings.json:1]: this is not supposed to be in the remote repository

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM mcr.microsoft.com/dotnet/sdk:latest as build
```
Comment (pietrykovsky) [inline Dockerfile:1]: since we are working on dotnet 8.0, we should specify it in case of compatibility issues with future dotnet releases. same for the aspnet below

File: .vscode/settings.json
Comment (Milosz-cat) [reply .vscode/settings.json:1]: fixed

File: compose.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+version: '3.8'
+services:
+  # db:
```
Comment (pietrykovsky) [inline compose.yaml:3]: since this is going to be part of different pull request let's remove these comments to have clean commits

File: dataprotectionkeys/key-f2775ffb-7be2-4281-868c-e08023256253.xml
Comment (pietrykovsky) [reply dataprotectionkeys/key-f2775ffb-7be2-4281-868c-e08023256253.xml:1]: I think it would be best to fix it before we merge this into master branch

**Comment by pietrykovsky** [comment]
it seems that the dataprotectionkey is not automatically generated and after deletion the application throws exceptions:
```log
fail: Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery[7]
szczurtv-app-1  |       An exception was thrown while deserializing the token.
szczurtv-app-1  |       Microsoft.AspNetCore.Antiforgery.AntiforgeryValidationException: The antiforgery token could not be decrypted.
szczurtv-app-1  |        ---> System.Security.Cryptography.CryptographicException: The key {f2775ffb-7be2-4281-868c-e08023256253} was not found in the key ring. For more information go to https://aka.ms/aspnet/dataprotectionwarning
szczurtv-app-1  |          at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.UnprotectCore(Byte[] protectedData, Boolean allowOperationsOnRevokedKeys, UnprotectStatus& status)
szczurtv-app-1  |          at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.Unprotect(Byte[] protectedData)
szczurtv-app-1  |          at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
szczurtv-app-1  |          --- End of inner exception stack trace ---
szczurtv-app-1  |          at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
szczurtv-app-1  |          at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext)
```
